### DISCUSSION TITLE
Prevent form submit on Enter

### DISCUSSION THREAD

**Discussion by borex** [filamentphp/filament#6067] [Help]
Is there a solution that prevents submitting a form when pressing enter on a text box? Currently I couldn't find one, so this is what I came up:
I created copies for create-record.blade.php and edit-record.blade.php in my resources folder and setup the views for the edit and create records with:
`protected static string $view = 'filament.resources.model.create-record';`

In the blade views I have made the following changes:
`<x-filament::form wire:submit.prevent="create" onkeydown="return event.key != 'Enter';">`

I don't know if this is a good solution for my problem, but it works.

Also, it would be great if the Enter key press submit could be disabled by and option on the page or form. What do you think?

**Comment by VAZaitesev** [comment]
the same question

**Reply by jimarick** [reply]
Would also like this

**Reply by kismonguz** [reply]
That would be a nice QoL feature.

**Comment by kismonguz** [comment]
This solution will prevent the textarea component to accept enter key.

I found an another solution for this:
No need to override any view file, just simply override the Action button to act as a button not a submit:
In your "app/Filament/Resources/[...]Resource/Pages/Create[...].php" file you can override the behaviour of create button like this:
```php
    protected function getCreateFormAction(): Action
    {
        return parent::getCreateFormAction()->extraAttributes(['type' => 'button', 'wire:click' => 'create']);
    }
```
And in your "app/Filament/Resources/[...]Resource/Pages/Edit[...].php":
```php
    protected function getSaveFormAction(): Action
    {
        return parent::getSaveFormAction()->extraAttributes(['type' => 'button', 'wire:click' => 'save']);
    }
```


Note: If your form has contain very few fields (eg: only 1 text input, but didn't test a lot) the enter key still trigger submit. I think it is acceptable.

**Comment by aurawindsurfing** [comment]
@borex solution works as expected, disables enter on all wizard steps so user needs to use provided action buttons.

In my case I had to stop browser from submitting the form on receipt of OTP.
